### PR TITLE
Maintenance/migrate to gradle

### PR DESCRIPTION
**Pull request by nimakro**
This is quite a big PR let me explain:
Currently the plugin-project is just a normal intelliJ project and we have parts of `.idea` folder and the`*.iml`project file under vcs. This regularly breaks my setup and I spent way to much time getting everything working again. Time I could spend working on new features. Because of that I decided to migrate the project to the official gradle-plugin.
This has the following advantages:
* Everybody can simply clone the repo and just build and run it `gradle runIde`without adding dependencies manually to the project for example `kotlin-plugin` ....
* Easy publishing of the plugin with `gradle publishPlugin`
  @edvin be sure to add your credentials to the `gradle.properties` file and mark the file as *unchanged* to prevent adding your credentials to vcs `# git update-index --assume-unchanged gradle.properties`

To make gradle work I refactored the project to follow the default structure.
Kotlin files are under `src/main/kotlin` and the few remaining java files are under `src/main/java`

### CODE REVIEW

**Comment by nimakro** [comment]
@edvin Not sure why we get a conflict here since I did a rebase of upstream/master let me know if I need to chanche anything

**Comment by edvin** [comment]
Fantastic initiative, thank you @nimakro ! I will try to merge it tonight :)

**Comment by nimakro** [comment]
Hey @edvin you are very welcome... I just saw that @SiebelsTim submitted a PR as well so maybe you should review that one first and them I will rebase my PR gain with master, since I introduced a new folder structure...I also have a fix for the color annotator in the pipline.... as soon as we have that merged I think we should do a release (maybe you can try the gradle-plugin)?

**Comment by SiebelsTim** [comment]
I only added new files. I should be able to rebase my PR onto this. Don't worry about me :)

File: src/main/kotlin/no/tornado/tornadofx/idea/icons/PluginIcons.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package no.tornado.tornadofx.idea.icons
+
+import com.intellij.openapi.util.IconLoader
+
+object PluginIcons {
+    val ACTION = IconLoader.getIcon("/icons/action.png");
```
Comment (SiebelsTim) [inline src/main/kotlin/no/tornado/tornadofx/idea/icons/PluginIcons.kt]: `@JvmField`?

Also, unnecessary semicolon

File: src/main/java/no/tornado/tornadofx/idea/TornadoFXModuleType.java [hunk 2]
```diff
@@ -23,11 +24,11 @@ public String getDescription() {
 	}
 
 	public Icon getBigIcon() {
-		return ACTION;
+		return PluginIcons.INSTANCE.getACTION();
```
Comment (SiebelsTim) [inline src/main/java/no/tornado/tornadofx/idea/TornadoFXModuleType.java]: The `@JvmField` would make this way nicer

    return PluginIcons.ACTION;

**Comment by SiebelsTim** [comment]
Works fine for me. Nevertheless, I've added a comment.

The merge conflict is real though.

```
	deleted by them: resources/projectTemplates/tornadofx-gradle-project.zip
	deleted by them: resources/projectTemplates/tornadofx-maven-osgi-ds-project.zip
	deleted by them: resources/projectTemplates/tornadofx-maven-osgi-project.zip
	deleted by them: resources/projectTemplates/tornadofx-maven-project.zip
```

That probably happened because you merged my `padding`-PR yourself. And now it's diverged from master.
A rebase should fix this:

    git checkout maintenance/migrate-to-gradle && git rebase master && git push fork maintenance/migrate-to-gradle -f

File: src/main/kotlin/no/tornado/tornadofx/idea/icons/PluginIcons.kt (hunk 1 above)
Comment (nimakro) [reply src/main/kotlin/no/tornado/tornadofx/idea/icons/PluginIcons.kt]: You are right will fix this later ;-)

File: src/main/java/no/tornado/tornadofx/idea/TornadoFXModuleType.java (hunk 2 above)
Comment (nimakro) [reply src/main/java/no/tornado/tornadofx/idea/TornadoFXModuleType.java]: Yes, I wonder why I missed that ;-)

**Comment by nimakro** [comment]
@SiebelsTim Mhm I will check again, if I remember correct I rebased with master, I wonder how your PR got in there ;-)... I investigate ...
@edvin Just wait with merging till I fixed commit the changes...
@SiebelsTim By the way thank you for all you work you are contributing to the plugin, it's awesome to get some more people on board.

**Comment by nimakro** [comment]
@edvin I think we are ready to merge. @SiebelsTim I added your suggestions from the code review, thanks again ;-)

**Comment by SiebelsTim** [comment]
I broke your PR again :x

**Comment by edvin** [comment]
As far as I can tell, this looks good and all we need to do is to resolve the `plugin.xml` conflict before merging. @nimakro - merge when you're ready :) @SiebelsTim I've added you as collaborator with write access :)

**Comment by nimakro** [comment]
So I think every think should be working again ;-), so @edvin you can go ahead and merge.
@SiebelsTim welcome on board and once again thank's for your contributions ;-)

**Comment by edvin** [comment]
Great job guys! By the end of the week we should release the next TornadoFX version along with the plugin :)

**Pull request merged**